src/TLuaInterpreter.cpp

```diff
@@ -1133,17 +1130,15 @@ int TLuaInterpreter::loadRawFile(lua_State* L)
 int TLuaInterpreter::setProfileIcon(lua_State* L)
 {
     QString iconPath;
+
     if (!lua_isstring(L, 1)) {
         lua_pushfstring(L, "setProfileIcon: bad argument #1 type (icon file path expected, got %s!)",
                         luaL_typename(L, 1));
         return lua_error(L);
-    } else {
-        iconPath = QString::fromUtf8(lua_tostring(L, 1));
-        if (iconPath.isEmpty()) {
-            lua_pushnil(L);
-            lua_pushstring(L, "a blank string is not a valid icon file location");
-            return 2;
-        }
+    } else if (iconPath = QString::fromUtf8(lua_tostring(L, 1)); iconPath.isEmpty()) {
```

Comment:
As above this is *also* a Clazy `else after return` and we want the value beyond the conditional which means this can be better written as:
```
int TLuaInterpreter::setProfileIcon(lua_State* L)
{
    if (!lua_isstring(L, 1)) {
        lua_pushfstring(L, "setProfileIcon: bad argument #1 type (icon file path expected, got %s!)",
                        luaL_typename(L, 1));
        return lua_error(L);
    }
    QString iconPath{QString::fromUtf8(lua_tostring(L, 1))};
    if (iconPath.isEmpty()) {
        lua_pushnil(L);
        lua_pushstring(L, "a blank string is not a valid icon file location");
        return 2;
    }
 ...
```

and there is no need to get all C++17 ish to do it... :stuck_out_tongue_winking_eye: